pkg/operator/operator_config.go

```diff
@@ -370,9 +425,260 @@ func makeAlertManagerConfigs(spec *monitoringv1alpha1.AlertingSpec) ([]yaml.MapS
 		if am.Scheme != "" {
 			cfg = append(cfg, yaml.MapItem{Key: "scheme", Value: am.Scheme})
 		}
-		// TODO(pintohutch): fill the rest out.
 
+		// Authorization.
+		// Default to Bearer for authorization type.
+		if am.Authorization != nil {
+			authCfg := yaml.MapSlice{}
+			if t := am.Authorization.Type; t != "" {
+				authCfg = append(authCfg, yaml.MapItem{Key: "type", Value: strings.TrimSpace(t)})
+			}
+			if c := am.Authorization.Credentials; c != nil {
+				b, err := getSecretKeyBytes(ctx, r.client, c)
+				if err != nil {
+					return configs, err
+				}
+				authCfg = append(authCfg, yaml.MapItem{Key: "credentials", Value: string(b)})
```

Comment:
Since we are now inlining secret contents into the config, the final rule-evaluator config should be stored in a secret rather than a config map. Even though this unfortunately makes debugging more difficult.